### PR TITLE
Dlampriadis/masking policy overview

### PR DESCRIPTION
**Pull request by DLampriadis**
## Summary
Should produce an overview of the masking policies applied to the tables.

## Does this PR refer to any issue?
#22 

### Files
- masking_policy_overview.sql

### Procedure
- masking_policy_overview()

### Tables
- masking_policy_overview_results

## I made sure that...
- [x] The files are named `{health_check_name}.sql/py`, are saved in the proper directory and are well documented
- [x] The procedure is named `{health_check_name}` and is stored in the schema `health_checks.public`
- [x] The tables created are named `{health_check_name}[_extra_info]_[results or intermediate]` and are stored in the schema `health_checks.public`
- [x] I run the pre-commit hooks in all my commits

## Does my code code need to be tested?
Yes

## The reviewer should check
1. Run procedure
2. Select from created table
3. Give insights on how this would look like or used by the end user. Maybe just the table is not enough

### CODE REVIEW

File: snowflake-health-checks/src/libraries/procedures/masking_policy_overview.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                and deleted is null
+        )
+
+        select
+                policy_name
+            ,   concat(ref_database_name, '.', ref_schema_name, '.', ref_entity_name) as object_name
+            ,   ref_entity_domain as object_type
+            ,   ref_column_name as on_column
+            ,   owner.policy_owner as policy_owner
+        from SNOWFLAKE.ACCOUNT_USAGE.POLICY_REFERENCES final
```
Comment (wiesnseb) [inline snowflake-health-checks/src/libraries/procedures/masking_policy_overview.sql:27]: why capital amigo?

**Review by wiesnseb — CHANGES_REQUESTED** [review]
see comment

**Comment by DLampriadis** [comment]
Done, ready for review

File: snowflake-health-checks/src/libraries/procedures/masking_policy_overview.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            ,   ref_column_name as on_column
+            ,   owner.policy_owner as policy_owner
+        from snowflake.account_usage.policy_references final
+        join owner
+            on owner.policy_id = final.policy_id
+;
+case
+    when (select count(*) from health_checks.public.masking_policy_overview_results) > 0
+    then
+        return 'There objects in the account with masking policies applied on them. Have an overview here: health_checks.public.masking_policy_overview_results'
```
Comment (EloiSanchez) [inline snowflake-health-checks/src/libraries/procedures/masking_policy_overview.sql:34]: Fix typo here: 'There **are** objects in the account ...'

**Review by EloiSanchez — APPROVED** [review]
Fix the typo then it will be fine for merging already.

**Review by EloiSanchez — APPROVED** [review]
Nice

**Review by wiesnseb — APPROVED** [review]
all good now

**Pull request merged**